### PR TITLE
Add indexed fields to span.context.http.

### PR DESCRIPTION
**Pull request by simitt**
Add `status_code` and `method` to `span.context.http` and index them. The new fields are added to v1 and v2 as they are an extension of the current schema. 

There wasn't a big discussion about whether or not to index the fields, but they both seem reasonable to index.

The changes do not automatically show up in the UI, so this would require some work from the @elastic/apm-ui  team. 

implements https://github.com/elastic/apm-server/issues/1435

### CODE REVIEW

File: docs/fields.asciidoc [hunk 1]
```diff
@@ -93,6 +93,27 @@ A flat mapping of user-defined tags with string values.
...
+
+
+--
+
+*`context.http.status_code`*::
++
+--
+type: long
+
+The status code of the http request.
```
Comment (jalvz) [inline docs/fields.asciidoc]: nitpick: this is actually the HTTP response.

i am also seeing that we write indifferently HTTP and http, maybe you can use this change to grep and replace to all caps everywhere.

**Comment by simitt** [comment]
@elastic/apm-server any concerns about indexing those fields? Otherwise I go forward with it.

**Comment by simitt** [comment]
After another discussion with @eyalkoren I went ahead with only indexing `context.http.status_code` but not `context.http.method`, as this doesn't seem to be necessary for now.

**Comment by simitt** [comment]
jenkins, retest this please

**Pull request merged**